modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/odata/filter/expression/SOSQLExpressionVisitorImpl.java

```diff
@@ -152,6 +153,10 @@ public Object visitLiteralExpression(LiteralExpression literalExpression)
 				StringPool.APOSTROPHE);
 		}
 
+		if (Objects.equals(null, literalExpression.getType())) {
+			return null;
+		}
```

Comment:
Hey @jairmedeiros we can move this conditional to be checked first this way we won't need to verify if the liferay Expression.getType is not null in the conditional above.

Thank you!